Content.Server/Blob/BlobCoreSystem.cs

```diff
@@ -0,0 +1,338 @@
+using Content.Server.Fluids.EntitySystems;
+using Content.Server.GameTicking;
+using Content.Server.GameTicking.Rules.Components;
+using Content.Server.Mind;
+using Content.Server.Objectives;
+using Content.Server.Roles;
+using Content.Shared.Alert;
+using Content.Shared.Blob;
+using Content.Shared.Damage;
+using Content.Shared.Destructible;
+using Content.Shared.FixedPoint;
+using Content.Shared.Popups;
+using Content.Shared.Roles;
+using Content.Shared.Weapons.Melee;
+using Robust.Server.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+using Robust.Shared.Network;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Blob;
+
+public sealed class BlobCoreSystem : EntitySystem
+{
+    [Dependency] private readonly AlertsSystem _alerts = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly MindSystem _mindSystem = default!;
+    [Dependency] private readonly IChatManager _chatManager = default!;
+    [Dependency] private readonly AudioSystem _audioSystem = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IMapManager _mapManager = default!;
+    [Dependency] private readonly GameTicker _gameTicker = default!;
+    [Dependency] private readonly BlobObserverSystem _blobObserver = default!;
+    [Dependency] private readonly ExplosionSystem _explosionSystem = default!;
+    [Dependency] private readonly DamageableSystem _damageable = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<BlobCoreComponent, ComponentStartup>(OnStartup);
+        SubscribeLocalEvent<BlobCoreComponent, DestructionEventArgs>(OnDestruction);
+        SubscribeLocalEvent<BlobCoreComponent, DamageChangedEvent>(OnDamaged);
+        SubscribeLocalEvent<BlobCoreComponent, PlayerAttachedEvent>(OnPlayerAttached);
+    }
+
+    private void OnPlayerAttached(EntityUid uid, BlobCoreComponent component, PlayerAttachedEvent args)
+    {
+        var xform = Transform(uid);
+        if (!_mapManager.TryGetGrid(xform.GridUid, out var map))
+            return;
+
+        CreateBlobObserver(uid, args.Player.UserId, component);
+    }
+
+    public bool CreateBlobObserver(EntityUid blobCoreUid, NetUserId userId, BlobCoreComponent? core = null)
+    {
+        var xform = Transform(blobCoreUid);
+
+        if (!Resolve(blobCoreUid, ref core))
+            return false;
+
+        var blobRule = EntityQuery<BlobRuleComponent>().FirstOrDefault();
+
+        if (blobRule == null)
+        {
+            _gameTicker.StartGameRule("Blob", out var ruleEntity);
+            blobRule = Comp<BlobRuleComponent>(ruleEntity);
+        }
+
+        var observer = Spawn(core.ObserverBlobPrototype, xform.Coordinates);
+
+        core.Observer = observer;
+
+        if (!TryComp<BlobObserverComponent>(observer, out var blobObserverComponent))
```

Comment:
This is not an ignorable error, especially as none of the places that call CreateBlobObserver actually check if it succeeded (WHY??). Throw an exception.